Mage.Sets/src/mage/cards/a/AshiokWickedManipulator.java

```diff
@@ -0,0 +1,125 @@
+        this.addAbility(new LoyaltyAbility(
+                new LookLibraryAndPickControllerEffect(2, 1, PutCards.EXILED, PutCards.HAND),
+                1
+        ));
+
+        // -2: Create two 1/1 black Nightmare creature tokens with "At the beginning of combat on your turn, if a card was put into exile this turn, put a +1/+1 counter on this creature."
+        this.addAbility(new LoyaltyAbility(
+                new CreateTokenEffect(new AshiokWickedManipulatorNightmareToken(), 2),
+                -2
+        ), new CardsExiledThisTurnWatcher());
+
+        // -7: Target player exiles the top X cards of their library, where X is the total mana value of cards you own in exile.
+        Ability ability = new LoyaltyAbility(
+                new ExileCardsFromTopOfLibraryTargetEffect(
+                        TotalCardsExiledOwnedManaValue.instance,
+                        "target player"
+                ).setText("target player exiles the top X cards of their library, "
+                        + "where X is the total mana value of cards you own in exile"),
+                -7
+        );
+        ability.addTarget(new TargetPlayer());
+        ability.addHint(TotalCardsExiledOwnedManaValue.getHint());
+        this.addAbility(ability);
+    }
+
+    private AshiokWickedManipulator(final AshiokWickedManipulator card) {
+        super(card);
+    }
+
+    @Override
+    public AshiokWickedManipulator copy() {
+        return new AshiokWickedManipulator(this);
+    }
+}
+
+class AshiokWickedManipulatorReplacementEffect extends ReplacementEffectImpl {
+
+    AshiokWickedManipulatorReplacementEffect() {
+        super(Duration.WhileOnBattlefield, Outcome.Benefit);
+        staticText = "If you would pay life while your library has at least that many cards in it, "
+                + "exile that many cards from the top of your library instead.";
+    }
+
+    private AshiokWickedManipulatorReplacementEffect(final AshiokWickedManipulatorReplacementEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public AshiokWickedManipulatorReplacementEffect copy() {
+        return new AshiokWickedManipulatorReplacementEffect(this);
+    }
+
+    @Override
+    public boolean replaceEvent(GameEvent event, Ability source, Game game) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) {
+            return false;
+        }
+
+        Set<Card> cards = player.getLibrary().getTopCards(game, event.getAmount());
+        player.moveCardsToExile(cards, source, game, false, null, "");
+
+        return true;
+    }
+
+    @Override
+    public boolean checksEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.PAY_LIFE;
+    }
+
+    @Override
+    public boolean applies(GameEvent event, Ability source, Game game) {
+        UUID playerId = source.getControllerId();
+        if (playerId == null || !event.getPlayerId().equals(playerId)) {
```

Comment:
No need to null check an object if you're only using it inside the equals method and not calling any methods on it